broker/deprovision_endpoint.go

```diff
@@ -22,15 +23,27 @@ func (bkr *Broker) deprovision(instanceID structs.ClusterID, details brokerapi.D
 		return false, err
 	}
 
-	cluster, err := bkr.state.LoadCluster(instanceID)
+	clusterState, err := bkr.state.LoadCluster(instanceID)
 	if err != nil {
 		logger.Error("load-cluster.error", err)
 		return false, err
 	}
 
-	bkr.scheduler.StopCluster(cluster)
-	bkr.state.DeleteCluster(cluster.InstanceID)
-	bkr.router.RemoveClusterAssignment(cluster.InstanceID)
+	clusterModel := state.NewClusterStateModel(bkr.state, clusterState)
+	err = clusterModel.ResetClusterPlan()
```

Comment:
Why is this being called during provision?
Shouldn't this always be reset after having created a new ClusterModel?